numba/core/extending_hardware.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC, abstractmethod
```

Comment:
In the next PR, we should make sure this file can be imported in the top everywhere.

Reply:
Agree.